developer.md

```diff
@@ -131,3 +131,38 @@ There are two reasons to add a finalizer to a CR instance from its reconciler:
 2. If the instance needs specific cleanup actions. For example running a `Job`
 when the instance is being deleted. Note that deleting children CRs are
 automatic if `OwnerReferece` is set no explict delete is needed there.
+
+## Defaulting structs
+When a CRD has an optional struct field the defaulting of that field needs
+special care. Both the field with the struct type need a full default value
+defined and each individual subfields in the struct needs default value
+defined.
+
+Our common example is the `PasswordSelectors` struct.
+
+```golang
+type PlacementAPISpec struct {
+    //...
+    // +kubebuilder:validation:Optional
```

Comment:
maybe fix this to align with the next line?

Reply:
I blame vscode not showing the tab and the spaces differently. Now I replaced all with tabs.